x/concentrated-liquidity/pool.go

```diff
@@ -101,6 +101,41 @@ func (k Keeper) setPool(ctx sdk.Context, pool types.ConcentratedPoolExtension) e
 	return nil
 }
 
+func (k Keeper) GetPoolDenoms(ctx sdk.Context, poolId uint64) ([]string, error) {
+	concentratedPool, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return nil, err
+	}
+
+	denoms := []string{concentratedPool.GetToken0(), concentratedPool.GetToken1()}
+	return denoms, nil
+}
+
+func (k Keeper) CalculateSpotPrice(
+	ctx sdk.Context,
+	poolId uint64,
+	quoteAssetDenom string,
+	baseAssetDenom string,
+) (spotPrice sdk.Dec, err error) {
+	concentratedPool, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return sdk.Dec{}, err
+	}
+
+	sqrtPrice := concentratedPool.GetCurrentSqrtPrice().Power(2)
```

Comment:
This is no longer a sqrtPrice, just a price since we are squaring it.

Reply:
https://github.com/osmosis-labs/osmosis/pull/4299/commits/1d1761106a70069bfd63ffe1427cbad0ebf6e9ca